File: data-layer/network-layer/src/main/java/com/mohamadamin/cleantvmaze/network/api/ShowsAPI.java
package com.mohamadamin.cleantvmaze.network.api;

import com.mohamadamin.cleantvmaze.domain.entity.Episode;
import com.mohamadamin.cleantvmaze.domain.entity.Season;
import com.mohamadamin.cleantvmaze.domain.entity.Show;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

/**
 * @author MohamadAmin Mohamadi (mohammadi.mohamadamin@gmail.com) on 5/6/17.
 */

public interface ShowsAPI {

    @GET("/shows")
    Observable<List<Show>> getShows(@Query("page") int page);

    @GET("/shows/{id}")
    Observable<Show> getShow(@Path("id") int showId);

    @GET("/shows/{id}/episodes")
    Observable<List<Episode>> getEpisodes(@Path("id") int showId);

    @GET("/shows/{id}/seasons")
    Observable<List<Season>> getSeasons(@Path("id") int showId);

    @GET("/search/shows")
    Observable<List<Show>> searchShows(@Query("q") String query);

    @GET("/singlesearch/shows")
    Observable<Show> searchSingleShow(@Query("q") String query);

}

File: data-layer/network-layer/src/main/java/com/mohamadamin/cleantvmaze/network/api/ShowsAPI.kt
package com.mohamadamin.cleantvmaze.data.network

import com.mohamadamin.cleantvmaze.domain.entity.Episode
import com.mohamadamin.cleantvmaze.domain.entity.Season
import com.mohamadamin.cleantvmaze.domain.entity.Show
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import rx.Observable

/**
 * @author MohamadAmin Mohamadi (mohammadi.mohamadamin@gmail.com) on 5/1/17.
 */
interface ShowsAPI {

    @GET("/shows")
    fun getShows(@Query("page") page: Int): Observable<List<Show>>

    @GET("/shows/{id}")
    fun getShow(@Path("id") showId: Int): Observable<Show>

    @GET("/shows/{id}/episodes")
    fun getEpisodes(@Path("id") showId: Int): Observable<List<Episode>>

    @GET("/shows/{id}/seasons")
    fun getSeasons(@Path("id") showId: Int): Observable<List<Season>>

    @GET("/search/shows")
    fun searchShows(@Query("q") query: String): Observable<List<Show>>

    @GET("/singlesearch/shows")
    fun searchSingleShow(@Query("q") query: String): Observable<Show>

}